src/environmentd/src/lib.rs

```diff
@@ -360,11 +390,25 @@ pub async fn serve(config: Config) -> Result<Server, anyhow::Error> {
     if let (Some(segment_client), Some(frontegg)) = (segment_client, config.frontegg) {
         telemetry::start_reporting(telemetry::Config {
             segment_client,
-            adapter_client,
+            adapter_client: adapter_client.clone(),
             organization_id: frontegg.tenant_id(),
         });
     }
 
+    // If system_parameter_frontend is present, start the system_parameter_sync loop.
+    if let Some(system_parameter_frontend) = system_parameter_frontend {
+        let system_parameter_backend = SystemParameterBackend::new(adapter_client).await?;
+        task::spawn(
+            || "system_parameter_sync",
+            AssertUnwindSafe(system_parameter_sync(
+                system_parameter_frontend,
+                system_parameter_backend,
+                config.config_sync_loop_interval,
+            ))
+            .ore_catch_unwind(),
```

Comment:
I tried appending a `map` to report any panics to Sentry, like that:

```rust
            .ore_catch_unwind()
            .map(|result| {
                if let Err(error) = result {
                    if let Ok(error) = error.downcast::<&str>() {
                        tracing::error!("unexpected {}", error);
                    } else {
                        tracing::error!("unexpected error");
                    }
                }
            }),
```

but I didn't see anything after hard-coding a panic in `system_parameter_sync`.

Reply:
And you tried this with the `--sentry-dsn` argument, or whatever it is, that you need to actually report to Sentry?

Reply:
No, I meant that if I do this the error in stdout/stderr (whereas I see `tracing` messages from the main loop).